packages/lesswrong/lib/vulcan-lib/getCollection.ts

```diff
@@ -50,13 +51,17 @@ export const isValidCollectionName = (name: string): name is CollectionNameStrin
 
 // Add a collection to Collections and collectionsByName. Should only be called
 // from createCollection.
-export const registerCollection = (collection: CollectionBase<any>): void => {
-  Collections.push(collection);
-  collectionsByName[collection.collectionName] = collection;
+export const registerCollection = <N extends CollectionNameString>(
+  collection: CollectionBase<N>
+): void => {
+  Collections.push(collection as CollectionBase<CollectionNameString>);
+  collectionsByName[collection.collectionName] = collection as unknown as CollectionsByName[N];
```

Comment:
This seems less than ideal, although I can't immediately come up with a better way of doing it (presumably you also tried)